--- celeste/Decompiled//Celeste.Pico8/Emulator.cs	2024-12-27 20:20:55.847989048 -0800
+++ celeste/Celeste.Pico8/Emulator.cs	2024-12-27 20:20:57.194754924 -0800
@@ -29,7 +29,7 @@
 
 	private bool skipFrame = true;
 
-	private EventInstance bgSfx;
+	private EventInstance? bgSfx;
 
 	private VirtualRenderTarget buffer;
 
@@ -41,7 +41,7 @@
 
 	private float pauseFade;
 
-	private EventInstance snapshot;
+	private EventInstance? snapshot;
 
 	private MTexture picoBootLogo;
 
