--- celeste/Decompiled/Celeste.Pico8/Emulator.cs
+++ celeste/Celeste.Pico8/Emulator.cs
@@ -29,7 +29,7 @@
 
 	private bool skipFrame = true;
 
-	private EventInstance bgSfx;
+	private EventInstance? bgSfx;
 
 	private VirtualRenderTarget buffer;
 
@@ -41,7 +41,7 @@
 
 	private float pauseFade;
 
-	private EventInstance snapshot;
+	private EventInstance? snapshot;
 
 	private MTexture picoBootLogo;
 
