--- celeste/Decompiled/FMOD/Memory.cs
+++ celeste/FMOD/Memory.cs
@@ -7,7 +7,7 @@
 {
 	public static RESULT Initialize(IntPtr poolmem, int poollen, MEMORY_ALLOC_CALLBACK useralloc, MEMORY_REALLOC_CALLBACK userrealloc, MEMORY_FREE_CALLBACK userfree, MEMORY_TYPE memtypeflags = MEMORY_TYPE.ALL)
 	{
-		return FMOD_Memory_Initialize(poolmem, poollen, useralloc, userrealloc, userfree, memtypeflags);
+		return WRAP_FMOD_Memory_Initialize(poolmem, poollen, useralloc, userrealloc, userfree, memtypeflags);
 	}
 
 	public static RESULT GetStats(out int currentalloced, out int maxalloced)
@@ -17,12 +17,12 @@
 
 	public static RESULT GetStats(out int currentalloced, out int maxalloced, bool blocking)
 	{
-		return FMOD_Memory_GetStats(out currentalloced, out maxalloced, blocking);
+		return WRAP_FMOD_Memory_GetStats(out currentalloced, out maxalloced, blocking);
 	}
 
 	[DllImport("fmod")]
-	private static extern RESULT FMOD_Memory_Initialize(IntPtr poolmem, int poollen, MEMORY_ALLOC_CALLBACK useralloc, MEMORY_REALLOC_CALLBACK userrealloc, MEMORY_FREE_CALLBACK userfree, MEMORY_TYPE memtypeflags);
+	private static extern RESULT WRAP_FMOD_Memory_Initialize(IntPtr poolmem, int poollen, MEMORY_ALLOC_CALLBACK useralloc, MEMORY_REALLOC_CALLBACK userrealloc, MEMORY_FREE_CALLBACK userfree, MEMORY_TYPE memtypeflags);
 
 	[DllImport("fmod")]
-	private static extern RESULT FMOD_Memory_GetStats(out int currentalloced, out int maxalloced, bool blocking);
+	private static extern RESULT WRAP_FMOD_Memory_GetStats(out int currentalloced, out int maxalloced, bool blocking);
 }
