--- celeste/Decompiled//Celeste/AudioState.cs	2024-12-27 20:20:54.907919827 -0800
+++ celeste/Celeste/AudioState.cs	2024-12-27 20:20:57.178087028 -0800
@@ -47,7 +47,7 @@
 			}
 			if (flag)
 			{
-				Audio.CurrentMusicEventInstance.start();
+				Audio.CurrentMusicEventInstance.Value.start();
 			}
 		}
 		flag = Audio.SetAmbience(Ambience.Event, startPlaying: false);
@@ -61,7 +61,7 @@
 		}
 		if (flag)
 		{
-			Audio.CurrentAmbienceEventInstance.start();
+			Audio.CurrentAmbienceEventInstance.Value.start();
 		}
 	}
 
