--- celeste/Decompiled/Celeste/AudioState.cs
+++ celeste/Celeste/AudioState.cs
@@ -47,7 +47,7 @@
 			}
 			if (flag)
 			{
-				Audio.CurrentMusicEventInstance.start();
+				Audio.CurrentMusicEventInstance.Value.start();
 			}
 		}
 		flag = Audio.SetAmbience(Ambience.Event, startPlaying: false);
@@ -61,7 +61,7 @@
 		}
 		if (flag)
 		{
-			Audio.CurrentAmbienceEventInstance.start();
+			Audio.CurrentAmbienceEventInstance.Value.start();
 		}
 	}
 
