--- celeste/Decompiled//Celeste/BadelineBoost.cs	2024-12-27 20:20:54.607897742 -0800
+++ celeste/Celeste/BadelineBoost.cs	2024-12-27 20:20:57.168086292 -0800
@@ -43,7 +43,7 @@
 
 	private SoundSource relocateSfx;
 
-	public FMOD.Studio.EventInstance Ch9FinalBoostSfx;
+	public FMOD.Studio.EventInstance? Ch9FinalBoostSfx;
 
 	public BadelineBoost(Vector2[] nodes, bool lockCamera, bool canSkip = false, bool finalCh9Boost = false, bool finalCh9GoldenBoost = false, bool finalCh9Dialog = false)
 		: base(nodes[0])
