--- celeste/Decompiled//Celeste/CassetteBlockManager.cs	2024-12-27 20:20:54.441218807 -0800
+++ celeste/Celeste/CassetteBlockManager.cs	2024-12-27 20:20:57.161419134 -0800
@@ -22,9 +22,9 @@
 
 	private int beatIndexOffset;
 
-	private EventInstance sfx;
+	private EventInstance? sfx;
 
-	private EventInstance snapshot;
+	private EventInstance? snapshot;
 
 	public CassetteBlockManager()
 	{
@@ -142,13 +142,13 @@
 				beatIndex = 0;
 				if (!isLevelMusic)
 				{
-					sfx.start();
+					sfx.Value.start();
 				}
 			}
 		}
 		if (leadBeats <= 0)
 		{
-			sfx.setParameterValue("sixteenth_note", GetSixteenthNote());
+			sfx.Value.setParameterByName("sixteenth_note", GetSixteenthNote());
 		}
 	}
 
