--- celeste/Decompiled/Celeste/Celeste.cs
+++ celeste/Celeste/Celeste.cs
@@ -1,6 +1,5 @@
 using System;
 using System.Diagnostics;
-using System.IO;
 using System.Threading;
 using Celeste.Pico8;
 using Microsoft.Xna.Framework;
@@ -56,7 +55,7 @@
 
 	public static readonly AppId_t SteamID = new AppId_t(504230u);
 
-	private static int _mainThreadId;
+	public static int _mainThreadId;
 
 	public static Vector2 TargetCenter => new Vector2(1920f, 1080f) / 2f;
 
@@ -83,9 +82,11 @@
 			Engine.ViewPadding = Settings.Instance.ViewportPadding;
 		}
 		Settings.Instance.ApplyScreen();
+		if (global::Program.firstLaunch) {
 		SFX.Initialize();
 		Tags.Initialize();
 		Input.Initialize();
+		}
 		Engine.Commands.Enabled = PlayMode == PlayModes.Debug;
 		Engine.Scene = new GameLoader();
 	}
@@ -165,6 +166,7 @@
 		}
 	}
 
+	/*
 	private static void Main(string[] args)
 	{
 		Celeste celeste;
@@ -240,6 +242,7 @@
 		celeste.Dispose();
 		Audio.Unload();
 	}
+*/
 
 	public static void ReloadAssets(bool levels, bool graphics, bool hires, AreaKey? area = null)
 	{
@@ -271,18 +274,6 @@
 
 	private static void CallProcess(string path, string args = "", bool createWindow = false)
 	{
-		Process process = new Process();
-		process.StartInfo = new ProcessStartInfo
-		{
-			FileName = path,
-			WorkingDirectory = Path.GetDirectoryName(path),
-			RedirectStandardOutput = false,
-			CreateNoWindow = !createWindow,
-			UseShellExecute = false,
-			Arguments = args
-		};
-		process.Start();
-		process.WaitForExit();
 	}
 
 	public static bool PauseAnywhere()
