--- celeste/Decompiled//Celeste/CS01_Ending.cs	2024-12-26 18:08:59.664505735 -0800
+++ celeste/Celeste/CS01_Ending.cs	2024-12-26 19:34:54.250454350 -0800
@@ -66,7 +66,7 @@
 		yield return 4f;
 		BirdNPC bird = new BirdNPC(player.Position + new Vector2(88f, -200f), BirdNPC.Modes.None);
 		base.Scene.Add(bird);
-		EventInstance instance = Audio.Play("event:/game/general/bird_in", bird.Position);
+		EventInstance? instance = Audio.Play("event:/game/general/bird_in", bird.Position);
 		bird.Facing = Facings.Left;
 		bird.Sprite.Play("fall");
 		Vector2 from = bird.Position;