--- celeste/Decompiled/Celeste/CS10_HubIntro.cs
+++ celeste/Celeste/CS10_HubIntro.cs
@@ -12,7 +12,7 @@
 	{
 		private Sprite sprite;
 
-		public EventInstance sfx;
+		public EventInstance? sfx;
 
 		public Bird(Vector2 position)
 		{
@@ -122,7 +122,7 @@
 			{
 				if (!@lock.Visible && level.Camera.Y + 90f > @lock.Y - 20f)
 				{
-					sfxs.Add(Audio.Play(sfx + doorIndex, @lock.Center));
+					sfxs.Add(Audio.Play(sfx + doorIndex, @lock.Center).Value);
 					@lock.Appear();
 					Input.Rumble(RumbleStrength.Strong, RumbleLength.Medium);
 					doorIndex++;
