--- celeste/Decompiled//Celeste/Level.cs	2024-12-27 20:20:55.867990521 -0800
+++ celeste/Celeste/Level.cs	2024-12-27 20:20:57.184754187 -0800
@@ -182,11 +182,11 @@
 
 	private bool endingChapterAfterCutscene;
 
-	public static EventInstance DialogSnapshot;
+	public static EventInstance? DialogSnapshot;
 
-	private static EventInstance PauseSnapshot;
+	private static EventInstance? PauseSnapshot;
 
-	private static EventInstance AssistSpeedSnapshot;
+	private static EventInstance? AssistSpeedSnapshot;
 
 	private static int AssistSpeedSnapshotValue = -1;
 
