--- celeste/Decompiled//Celeste/MenuOptions.cs	2024-12-27 20:20:55.231276968 -0800
+++ celeste/Celeste/MenuOptions.cs	2024-12-27 20:20:57.191421345 -0800
@@ -15,9 +15,9 @@
 
 	private static TextMenu.Item viewport;
 
-	private static EventInstance snapshot;
+	private static EventInstance? snapshot;
 
-	public static TextMenu Create(bool inGame = false, EventInstance snapshot = null)
+	public static TextMenu Create(bool inGame = false, EventInstance? snapshot = null)
 	{
 		MenuOptions.inGame = inGame;
 		MenuOptions.snapshot = snapshot;
