--- celeste/Decompiled/Celeste/MenuOptions.cs
+++ celeste/Celeste/MenuOptions.cs
@@ -15,9 +15,9 @@
 
 	private static TextMenu.Item viewport;
 
-	private static EventInstance snapshot;
+	private static EventInstance? snapshot;
 
-	public static TextMenu Create(bool inGame = false, EventInstance snapshot = null)
+	public static TextMenu Create(bool inGame = false, EventInstance? snapshot = null)
 	{
 		MenuOptions.inGame = inGame;
 		MenuOptions.snapshot = snapshot;
