--- celeste/Decompiled//Celeste/MoveBlock.cs	2024-12-27 20:20:55.341285069 -0800
+++ celeste/Celeste/MoveBlock.cs	2024-12-27 20:20:57.191421345 -0800
@@ -535,7 +535,7 @@
 				yield return null;
 			}
 			Collidable = true;
-			EventInstance instance = Audio.Play("event:/game/04_cliffside/arrowblock_reform_begin", debris[0].Position);
+			EventInstance instance = Audio.Play("event:/game/04_cliffside/arrowblock_reform_begin", debris[0].Position).Value;
 			MoveBlock moveBlock3 = this;
 			Coroutine component;
 			Coroutine routine = (component = new Coroutine(SoundFollowsDebrisCenter(instance, debris)));
