--- celeste/Decompiled//Celeste/OuiAssistMode.cs	2024-12-27 20:20:54.877917618 -0800
+++ celeste/Celeste/OuiAssistMode.cs	2024-12-27 20:20:57.174753450 -0800
@@ -42,7 +42,7 @@
 
 	private float rightArrowEase;
 
-	private EventInstance mainSfx;
+	private EventInstance? mainSfx;
 
 	private const float textScale = 0.8f;
 
@@ -96,7 +96,7 @@
 		}
 		if (mainSfx != null)
 		{
-			mainSfx.release();
+			mainSfx.Value.release();
 		}
 		pages.Clear();
 		Visible = false;
