--- celeste/Decompiled//Celeste/Player.cs	2024-12-26 18:09:02.691223392 -0800
+++ celeste/Celeste/Player.cs	2024-12-26 19:43:29.804995974 -0800
@@ -590,7 +590,7 @@
 
 	private List<ChaserStateSound> activeSounds = new List<ChaserStateSound>();
 
-	private EventInstance idleSfx;
+	private EventInstance? idleSfx;
 
 	public bool MuffleLanding;
 
@@ -672,7 +672,7 @@
 
 	private const float SpacePhysicsMult = 0.6f;
 
-	private EventInstance conveyorLoopSfx;
+	private EventInstance? conveyorLoopSfx;
 
 	private const float WallBoosterSpeed = -160f;
 
@@ -3912,8 +3912,8 @@
 	{
 		if (conveyorLoopSfx != null)
 		{
-			conveyorLoopSfx.setParameterValue("end", 1f);
-			conveyorLoopSfx.release();
+			conveyorLoopSfx.Value.setParameterByName("end", 1f);
+			conveyorLoopSfx.Value.release();
 			conveyorLoopSfx = null;
 		}
 		wallSpeedRetentionTimer = 0f;
@@ -3987,8 +3987,8 @@
 			wallBoosting = false;
 			if (conveyorLoopSfx != null)
 			{
-				conveyorLoopSfx.setParameterValue("end", 1f);
-				conveyorLoopSfx.release();
+				conveyorLoopSfx.Value.setParameterByName("end", 1f);
+				conveyorLoopSfx.Value.release();
 				conveyorLoopSfx = null;
 			}
 			float num = 0f;
@@ -6277,7 +6277,7 @@
 		}
 	}
 
-	public EventInstance Play(string sound, string param = null, float value = 0f)
+	public EventInstance? Play(string sound, string param = null, float value = 0f)
 	{
 		float value2 = 0f;
 		if (base.Scene is Level { Raining: not false })