--- celeste/Decompiled/Celeste/SoundSource.cs
+++ celeste/Celeste/SoundSource.cs
@@ -15,7 +15,7 @@
 
 	public bool RemoveOnOneshotEnd;
 
-	private EventInstance instance;
+	private EventInstance? instance;
 
 	private bool is3D;
 
@@ -33,7 +33,7 @@
 		{
 			if (instance != null)
 			{
-				instance.getPlaybackState(out var state);
+				instance.Value.getPlaybackState(out var state);
 				if (state == PLAYBACK_STATE.PLAYING || state == PLAYBACK_STATE.STARTING || state == PLAYBACK_STATE.SUSTAINING)
 				{
 					return true;
@@ -71,12 +71,13 @@
 	{
 		Stop();
 		EventName = path;
-		EventDescription eventDescription = Audio.GetEventDescription(path);
+		EventDescription? eventDescription = Audio.GetEventDescription(path);
 		if (eventDescription != null)
 		{
-			eventDescription.createInstance(out instance);
-			eventDescription.is3D(out is3D);
-			eventDescription.isOneshot(out isOneshot);
+			eventDescription.Value.createInstance(out var instanceNotNull);
+			instance = instanceNotNull;
+			eventDescription.Value.is3D(out is3D);
+			eventDescription.Value.isOneshot(out isOneshot);
 		}
 		if (instance != null)
 		{
@@ -91,9 +92,9 @@
 			}
 			if (param != null)
 			{
-				instance.setParameterValue(param, value);
+				instance.Value.setParameterByName(param, value);
 			}
-			instance.start();
+			instance.Value.start();
 			Playing = true;
 		}
 		return this;
@@ -103,7 +104,7 @@
 	{
 		if (instance != null)
 		{
-			instance.setParameterValue(param, value);
+			instance.Value.setParameterByName(param, value);
 		}
 		return this;
 	}
@@ -112,7 +113,7 @@
 	{
 		if (instance != null)
 		{
-			instance.setPaused(paused: true);
+			instance.Value.setPaused(paused: true);
 		}
 		Playing = false;
 		return this;
@@ -122,10 +123,10 @@
 	{
 		if (instance != null)
 		{
-			instance.getPaused(out var paused);
+			instance.Value.getPaused(out var paused);
 			if (paused)
 			{
-				instance.setPaused(paused: false);
+				instance.Value.setPaused(paused: false);
 				Playing = true;
 			}
 		}
@@ -160,10 +161,10 @@
 		{
 			return;
 		}
-		instance.getPlaybackState(out var state);
+		instance.Value.getPlaybackState(out var state);
 		if (state == PLAYBACK_STATE.STOPPED)
 		{
-			instance.release();
+			instance.Value.release();
 			instance = null;
 			Playing = false;
 			if (RemoveOnOneshotEnd)
