--- celeste/Decompiled//Celeste/WaveDashTutorialMachine.cs	2024-12-27 20:20:54.831247516 -0800
+++ celeste/Celeste/WaveDashTutorialMachine.cs	2024-12-27 20:20:57.171419871 -0800
@@ -35,9 +35,9 @@
 
 	private float interactStartZoom;
 
-	private EventInstance snapshot;
+	private EventInstance? snapshot;
 
-	private EventInstance usingSfx;
+	private EventInstance? usingSfx;
 
 	private SoundSource signSfx;
 
@@ -184,7 +184,7 @@
 		if (usingSfx != null)
 		{
 			Audio.SetParameter(usingSfx, "end", 1f);
-			usingSfx.release();
+			usingSfx.Value.release();
 		}
 		if (presentation != null)
 		{
@@ -221,7 +221,7 @@
 		if (usingSfx != null)
 		{
 			Audio.SetParameter(usingSfx, "quit", 1f);
-			usingSfx.release();
+			usingSfx.Value.release();
 			usingSfx = null;
 		}
 		Audio.ReleaseSnapshot(snapshot);
