--- celeste/Decompiled/Celeste/Audio.cs
+++ celeste/Celeste/Audio.cs
@@ -172,10 +172,7 @@
 		}
 		CheckFmod(FMOD.Studio.System.create(out Audio.system));
 		Audio.system.getLowLevelSystem(out var system);
-		if (SDL.SDL_GetPlatform().Equals("Linux"))
-		{
-			FMOD_SDL_Register(system.getRaw());
-		}
+		system.setDSPBufferSize(2048, 8);
 		CheckFmod(Audio.system.initialize(1024, studioFlags, FMOD.INITFLAGS.NORMAL, IntPtr.Zero));
 		attributes3d.forward = new VECTOR
 		{
