--- celeste/Decompiled/Celeste/GameLoader.cs
+++ celeste/Celeste/GameLoader.cs
@@ -61,6 +61,10 @@
 	private void LoadThread()
 	{
 		MInput.Disabled = true;
+		GFX.Load();
+		MTN.Load();
+		GFX.LoadData();
+		MTN.LoadData();
 		Stopwatch stopwatch = Stopwatch.StartNew();
 		Audio.Init();
 		Audio.Banks.Master = Audio.Banks.Load("Master Bank", loadStrings: true);
@@ -72,10 +76,6 @@
 		Settings.Instance.ApplyVolumes();
 		audioLoaded = true;
 		Console.WriteLine(" - AUDIO LOAD: " + stopwatch.ElapsedMilliseconds + "ms");
-		GFX.Load();
-		MTN.Load();
-		GFX.LoadData();
-		MTN.LoadData();
 		Stopwatch stopwatch2 = Stopwatch.StartNew();
 		Fonts.Prepare();
 		Dialog.Load();
