--- ../decomp/Celeste/BadelineBoost.cs	2024-05-17 02:12:37.000000000 -0400
+++ celeste/Celeste/BadelineBoost.cs	2024-05-21 12:02:13.710070044 -0400
@@ -43,7 +43,7 @@
 
 		private SoundSource relocateSfx;
 
-		public FMOD.Studio.EventInstance Ch9FinalBoostSfx;
+		public EventInstance? Ch9FinalBoostSfx;
 
 		public BadelineBoost(Vector2[] nodes, bool lockCamera, bool canSkip = false, bool finalCh9Boost = false, bool finalCh9GoldenBoost = false, bool finalCh9Dialog = false)
 			: base(nodes[0])