--- ../decomp/Celeste/GameLoader.cs	2024-05-17 02:12:38.000000000 -0400
+++ celeste/Celeste/GameLoader.cs	2024-11-20 20:58:54.095741276 -0500
@@ -54,8 +54,10 @@
 			handler.Tag = Tags.HUD;
 			handler.Add(new Coroutine(IntroRoutine()));
 			activeThread = Thread.CurrentThread;
-			activeThread.Priority = ThreadPriority.Lowest;
-			RunThread.Start(LoadThread, "GAME_LOADER", highPriority: true);
+            // Unsupported in WASM
+			// activeThread.Priority = ThreadPriority.Lowest;
+			//RunThread.Start(LoadThread, "GAME_LOADER", highPriority: true);
+            LoadThread();
 		}
 
 		private void LoadThread()
@@ -160,7 +162,7 @@
 				}
 			}
 			opening.Dispose();
-			activeThread.Priority = ThreadPriority.Normal;
+			//activeThread.Priority = ThreadPriority.Normal;
 			MInput.Disabled = false;
 			Engine.Scene = new OverworldLoader(Overworld.StartMode.Titlescreen, Snow);
 		}