--- ../decomp/Celeste/GameLoader.cs	2024-05-17 02:12:38.000000000 -0400
+++ celeste/Celeste/GameLoader.cs	2024-11-20 20:56:26.928729465 -0500
@@ -54,7 +54,8 @@
 			handler.Tag = Tags.HUD;
 			handler.Add(new Coroutine(IntroRoutine()));
 			activeThread = Thread.CurrentThread;
-			activeThread.Priority = ThreadPriority.Lowest;
+            // Unsupported in WASM
+			// activeThread.Priority = ThreadPriority.Lowest;
 			RunThread.Start(LoadThread, "GAME_LOADER", highPriority: true);
 		}
 
@@ -160,7 +161,7 @@
 				}
 			}
 			opening.Dispose();
-			activeThread.Priority = ThreadPriority.Normal;
+			//activeThread.Priority = ThreadPriority.Normal;
 			MInput.Disabled = false;
 			Engine.Scene = new OverworldLoader(Overworld.StartMode.Titlescreen, Snow);
 		}