--- ../decomp/Celeste/OuiFileSelect.cs	2024-05-17 02:12:39.000000000 -0400
+++ celeste/Celeste/OuiFileSelect.cs	2024-11-20 20:58:54.096741276 -0500
@@ -35,7 +35,8 @@
 						base.Scene.Remove(Slots[l]);
 					}
 				}
-				RunThread.Start(LoadThread, "FILE_LOADING");
+                LoadThread();
+				//RunThread.Start(LoadThread, "FILE_LOADING");
 				float elapsed = 0f;
 				while (!Loaded || elapsed < 0.5f)
 				{