--- ../decomp/Celeste/SummitVignette.cs	2024-05-17 02:12:39.000000000 -0400
+++ celeste/Celeste/SummitVignette.cs	2024-05-21 12:02:13.766071749 -0400
@@ -22,7 +22,8 @@
 		{
 			this.session = session;
 			session.Audio.Apply();
-			RunThread.Start(LoadCompleteThread, "SUMMIT_VIGNETTE");
+            LoadCompleteThread();
+			//RunThread.Start(LoadCompleteThread, "SUMMIT_VIGNETTE");
 		}
 
 		private void LoadCompleteThread()