--- ../decomp/Celeste/WaveDashTutorialMachine.cs	2024-05-17 02:12:38.000000000 -0400
+++ celeste/Celeste/WaveDashTutorialMachine.cs	2024-05-21 12:02:13.770071870 -0400
@@ -35,9 +35,9 @@
 
 		private float interactStartZoom;
 
-		private EventInstance snapshot;
+		private EventInstance? snapshot;
 
-		private EventInstance usingSfx;
+		private EventInstance? usingSfx;
 
 		private SoundSource signSfx;
 
@@ -184,7 +184,7 @@
 			if (usingSfx != null)
 			{
 				Audio.SetParameter(usingSfx, "end", 1f);
-				usingSfx.release();
+				usingSfx.Value.release();
 			}
 			if (presentation != null)
 			{
@@ -221,7 +221,7 @@
 			if (usingSfx != null)
 			{
 				Audio.SetParameter(usingSfx, "quit", 1f);
-				usingSfx.release();
+				usingSfx.Value.release();
 				usingSfx = null;
 			}
 			Audio.ReleaseSnapshot(snapshot);