--- celeste/Decompiled/Monocle/Engine.cs
+++ celeste/Monocle/Engine.cs
@@ -37,7 +37,7 @@
 
 	private int fpsCounter;
 
-	private static string AssemblyDirectory = Path.GetDirectoryName(Assembly.GetEntryAssembly().Location);
+	private static string AssemblyDirectory = "/";
 
 	public static Color ClearColor;
 
@@ -137,10 +137,9 @@
 			Graphics.PreferredBackBufferHeight = windowHeight;
 			Graphics.IsFullScreen = false;
 		}
-		base.Content.RootDirectory = "Content";
+		base.Content.RootDirectory = "libsdl/Content";
 		base.IsMouseVisible = false;
 		ExitOnEscapeKeypress = true;
-		GCSettings.LatencyMode = GCLatencyMode.SustainedLowLatency;
 	}
 
 	protected virtual void OnClientSizeChanged(object sender, EventArgs e)
